networkx/classes/reportviews.py

```diff
@@ -81,6 +95,107 @@
     to create 3-tuples and 4-tuples.
 
     The argument `nbunch` restricts edges to those incident to nodes in nbunch.
+
+Common usage patterns
+=====================
+
+NodeView / NodeDataView
+-----------------------
+
+    - Use ``G.nodes`` when you need membership tests, set-like operations, or to
+      look up a node's attribute dict with ``G.nodes[n]``.
+    - Use ``G.nodes(data=...)`` or ``G.nodes.data(attr_name, default=...)`` to
+      iterate node/data pairs or to extract a single attribute for all nodes.
+    - Use ``list(G)`` or ``for node in G:`` to iterate over nodes.
+
+Example:
+
+.. code-block:: python
+
+    >>> G = nx.path_graph(3)
+    >>> list(G.nodes)
+    [0, 1, 2]
+    >>> G.add_node(3, color="red")
+    >>> list(G.nodes.data("color", default=None))
+    [(0, None), (1, None), (2, None), (3, 'red')]
+
+EdgeView / EdgeDataView
+-----------------------
+
+    - Use ``G.edges`` to iterate or test membership for edges.
+    - Call ``G.edges(data=...)`` or ``G.edges(nbunch=..., data=..., keys=...)`` to
+      iterate edges with data, restrict to edges incident to a set of nodes, or
+      include multigraph keys.
+
+.. note::
+   Iteration of ``G.edges()`` yields node pairs as 2-tuples even for multigraphs.
+   Use ``G.edges(keys=True)`` to receive 3-tuples ``(u, v, key)`` for multigraphs.
```

Comment:
It'd be good to point out the difference between `G.edges` and `G.edges()` when using multigraphs.
```suggestion
.. note::
   Iteration of ``G.edges()`` yields node pairs as 2-tuples even for multigraphs.
   Iteration of `G.edges` yields 3-tuples for multigraphs and 2-tuples otherwise.
   You can also use ``G.edges(keys=True, data=True)`` to receive 4-tuples
   ``(u, v, key, data)`` for multigraphs.
```